linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -725,6 +725,37 @@ private[emitter] object CoreJSLib {
     private def defineDispatchFunctions(): Tree = {
       val instance = varRef("instance")
 
+      val methodsInRepresentativeClasses = globalKnowledge
+        .methodsInRepresentativeClasses()
+        .filter(_._2 != toStringMethodName) // toString() is special
+        .groupBy(_._2)
+        .map(kv => (kv._1, kv._2.map(_._1)))
+        .toList
+        .sortBy(_._1.nameString) // sort for determinism
```

Comment:
That's a fair argument. But why is a merged form of this computation slower than what we do now in `KnowledgeGuardian`? AFAICT it boils down to:
- Iterate over all method / class pairs
- Put them into a O(1) datastructure.

If we start talking about allocations etc. this can really go both ways IMO (because of tuples), and we should not speculate but measure.

Reply:
Hum, yes, I guess you're right; it's not clear where the benefit is. I've moved the computation to `KnowledgeGuardian` (although I did not filter out `toString` there, because that would not make sense).